x-pack/plugin/esql/qa/testFixtures/src/main/resources/categorize.csv-spec

```diff
@@ -1,13 +1,15 @@
 categorize
 required_capability: categorize
 
+# Drop the category ID, because it's non-deterministic
```

Comment:
I've re-added the `category` column and a `SORT message` inbetween. Thanks for the suggestion.

For my understanding: does the additional `SORT` step mean that every `CATEGORIZE` is happening on the coordinating node? I think it must be, otherwise this isn't working correctly, right?

Reply:
> For my understanding: does the additional SORT step mean that every CATEGORIZE is happening on the coordinating node? I think it must be, otherwise this isn't working correctly, right?

Yes. Your instincts for where we split are almost certainly right. We'll just run the agg after the sort. It's not ideal testing, but it's fine for now.

Reply:
Uhm, the CSV tests are executed in different ways. The `CsvTests` class, I think it executes like an unit test, like a single node. There are also tests like https://github.com/elastic/elasticsearch/blob/c1daf18bf5f5178d43dc17d3a3d1f5db9773098e/x-pack/plugin/esql/qa/server/multi-clusters/src/javaRestTest/java/org/elasticsearch/xpack/esql/ccq/MultiClusterSpecIT.java, which use a full cluster.
But now thinking about the categorize function and how it works, I don't really know how is the sort executed. And specially as this is the first grouping function depending on order, maybe it makes sense in fact to not check the category column...

Sorry, I think the original solution was correct 🙏 
Let's see if @nik9000 can confirm this

Reply:
[This failing test](https://gradle-enterprise.elastic.co/s/mh54vlgr4lvwk/tests/task/:x-pack:plugin:esql:qa:server:multi-node:javaRestTest/details/org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT/test%20%7Bcategorize.Categorize%20SYNC%7D?top-execution=1) suggests that data does not arrive at the categorizer in sorted order.

@nik9000 do you understand why?

I'll leave the test muted for now. Once we replace the (arbitrary) IDs by regexes, the output will be deterministic.